packages/website/src/docs/standard-library/http.md

```diff
@@ -294,3 +298,134 @@ namespace Pets {
+| "query"  | GET or HEAD request  |
+| "create" | POST or PUT request  |
+| "update" | PATCH or PUT request |
+| "delete" | DELETE request       |
+
+This allows a single logical Cadl model to be used as in the following example:
+
+```cadl
+model User {
+  name: string;
+  @visibility("read") id: string;
+  @visibility("create") password: string;
+}
+
+@route("/users")
+interface Users {
+  @post create(@path id: string, ...User): User;
+  @get get(@path id: string): User;
+}
+```
+
+There is a single logical user entity represented by the single Cadl type `User`, but the HTTP payload for this entity varies based on context. When returned in a response, the `id` property is included, but when sent in a request, it is not. Similarly, the `password` property is only included in create requests, but not present in responses.
+
+The OpenAPI v3 emitter will apply these visibilities automatically, without explicit use of `@withVisibility`, and it will generate separate schemas as necessary. However, another emitter such as one generating client code can see and preserve a single logical type and deal with these HTTP payload differences by means other than type proliferation.
+
+Modeling with logical entities rather than HTTP-specific shapes also keeps the Cadl spec decoupled from HTTP and REST and can allow the same spec to be used with multiple protocols.
+
+## Metadata
+
+The properties that designate content for the HTTP envelope (`@header`, `@path`, `@query`, `@statusCode`) rather than the content in an HTTP payload are often called "metadata".
+
+Metadata is determined to be applicable or inapplicable based on the context that it is used:
+
+| Context       | Applicability       |
+| ------------- | ------------------- |
+| `@query`      | request only        |
+| `@path`       | request only        |
+| `@statusCode` | response only       |
+| `@header`     | request or response |
+
+Additionally metadata that appears in an array element type always inapplicable.
+
+When metadata is deemed "inapplicable", for example, if a `@path` property is seen in a response, it becomes part of the payload instead.
+
+The handling of metadata applicability furthers the goal of keeping a single logical model in Cadl. For example, this defines a logical `User` entity that has a name, ID and password, but further annotates that the ID is sent in the HTTP path and the HTTP body in responses. Also, using automatically visibility as before, we further indicate that the password is only present in create requests.
+
+```cadl
+model User {
+  name: string;
+  @path id: string;
+  @visibility("create") password: string;
+}
+```
+
+Then, we can write operations in terms of the logical entity:
+
+```cadl
+@route("/users")
+interface Users {
+  @post create(...User): User;
+}
+```
+
+Abstractly, this expresses that a create operation that takes and returns a user. But concretely, at the HTTP protocol level, a create request and response look like this:
+
+```
```

Comment:
Think I need to fix something with the css of code blocks without a language.
<img width="1253" alt="image" src="https://user-images.githubusercontent.com/1031227/194374278-973b58ef-7ed6-4c97-9a9a-df16d56f77df.png">

Reply:
Double box and it looks bad if overlflows, fixed here https://github.com/microsoft/cadl/pull/1136

Reply:
Got it and thanks!